atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/backend/KeyValueServiceInstrumentation.java

```diff
@@ -43,27 +58,51 @@ public int getKeyValueServicePort() {
     public abstract KeyValueServiceConfig getKeyValueServiceConfig(InetSocketAddress addr);
     public abstract boolean canConnect(InetSocketAddress addr);
 
-    private static Map<String, KeyValueServiceInstrumentation> backendMap =
-            new TreeMap<>();
-
-    static {
-        addNewBackendType(new CassandraKeyValueServiceInstrumentation());
-        addNewBackendType(new PostgresKeyValueServiceInstrumentation());
+    public static void addNewBackendType(KeyValueServiceInstrumentation backend) {
+        if (!backendMap.containsKey(backend.getClassName())) {
+            classNames.put(backend.toString(), backend.getClassName());
+            backendMap.put(backend.getClassName(), backend);
+        }
     }
 
     public static KeyValueServiceInstrumentation forDatabase(String backend) {
-        return backendMap.get(backend);
+        if (classNames.containsKey(backend)) {
+            return backendMap.get(classNames.get(backend));
+        } else {
+            return forClass(backend);
+        }
     }
 
-    public static void addNewBackendType(KeyValueServiceInstrumentation backend) {
-        if (!backendMap.containsKey(backend.toString())) {
-            backendMap.put(backend.toString(), backend);
+    private static KeyValueServiceInstrumentation forClass(String className) {
+        if (!backendMap.containsKey(className)) {
+            addBackendFromClassName(className);
+        }
+        return backendMap.get(className);
+    }
+
+    private static void addBackendFromClassName(String className) {
+        try {
+            Class<?> clazz = Class.forName(className);
+            Constructor<?> constructor = clazz.getConstructor();
+            KeyValueServiceInstrumentation instance = (KeyValueServiceInstrumentation) constructor.newInstance();
+            addNewBackendType(instance);
+        } catch (Exception e) {
+            log.error("Exception trying to instantiate class {}:", className, e);
+            System.exit(1);
```

Comment:
Minor point: do we want to hard-fail if one of two classes cannot be instantiated?